pepsico/app_calc.py

```diff
@@ -143,3 +144,56 @@ def strftimeb2int(strftimeb):
     }
     strftimebint = strftimeb_all[strftimeb]
     return strftimebint
+
+
+def number_extreme_events_within_days(
+    daily_data, operator, threshold, window, dim="T"
+):
+    """Count extreme events
+    
+    Extreme events constitute cumulative `daily_data` over `window` -day
+    (or shorther) verifying the `operator` `threshold`
+    
+    Parameters
+    ----------
+    daily_data : DataArray
+        Array of daily data
+    operator : str
+        "lt", "le", "gt", "ge"
+        determines the condition to meet with respect to `threshold`
+    threshold : float
+        threshold to meet in `daily_data` units
+    window : int
+        maximum length of days to cumulate over to constitute an event
+    dim : str, optional
+        name of `daily_data` time dimension
+        
+    Returns
+    -------
+    DataArray
+        Array of count of extreme events
+    
+    Notes
+    -----
+    Is meant to be used to produce yearly seasonal time series
+    
+    Examples
+    --------
+    Number of rain events of >80mm in 2 days or less:
+    number_extreme_events_within_days(rain_daily_data_in_mm, "gt", 80, 2)    
+    """
+    count = 0
+    #Start with shortest events
+    for w in range(1, window+1):
+        for t in range(len(daily_data[dim])-(w-1)):
+            #Assert new event
+            new_event = getattr(op, operator)(
+                daily_data.isel({dim: slice(t, t+w)}).sum(dim), threshold
+            )
+            #Mask days having formed a new event so that they won't account again
+            #for following events (t loop) or longer events (w loop)
+            daily_data[{dim: slice(t, t+w)}] = (
```

Comment:
I strongly discourage you from modifying `daily_data`. The caller has no reason to suspect that calling this function will modify the contents of the array he provides as input. At the very least, such behavior would need to be documented, but people don't always read the documentation, so it's better just not to do it.

A less surprising way of implementing this algorithm would be to make a "deep copy" (see https://docs.xarray.dev/en/stable/generated/xarray.DataArray.copy.html) of `daily_data` and modify the copy rather than the original. But I doubt this is the best algorithm anyway, as I'll comment elsewhere.